tests/contrib/vertexai/test_vertexai.py

```diff
@@ -0,0 +1,499 @@
+import pytest
+
+from mock import PropertyMock, patch
+
+from tests.utils import override_global_config
+from tests.contrib.vertexai.utils import weather_tool
+from tests.contrib.vertexai.utils import _async_streamed_response
+from tests.contrib.vertexai.utils import _mock_completion_response
+from tests.contrib.vertexai.utils import _mock_completion_stream_chunk
+from tests.contrib.vertexai.utils import MOCK_COMPLETION_SIMPLE_1
+from tests.contrib.vertexai.utils import MOCK_COMPLETION_SIMPLE_2
+from tests.contrib.vertexai.utils import MOCK_COMPLETION_SIMPLE_3
+from tests.contrib.vertexai.utils import MOCK_COMPLETION_TOOL
+from tests.contrib.vertexai.utils import MOCK_COMPLETION_STREAM_CHUNKS
+from tests.contrib.vertexai.utils import MOCK_COMPLETION_TOOL_CALL_STREAM_CHUNKS
+
+def test_global_tags(vertexai, mock_client, mock_tracer):
+    """
+    When the global config UST tags are set
+        The service name should be used for all data
+        The env should be used for all data
+        The version should be used for all data
+    """
+    with patch.object(vertexai.generative_models.GenerativeModel, '_prediction_client', new_callable=PropertyMock) as mock_client_property:
+        mock_client_property.return_value = mock_client
```

Comment:
Seems like this setup code is repeated in every test, can we move it to the conftest.py as a part of the setup for the mock_client fixture?

Reply:
I found a way to do this within the vertexai fixture... should reduce the amount of repetitive patching code!